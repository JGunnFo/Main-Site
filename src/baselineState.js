
export const initialStateReducer={
    section: "main",
    transition:false
  };

export const initialStateTest={
    section: "main",
    transition:false
};
