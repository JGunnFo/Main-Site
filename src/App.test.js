import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from "react-redux";
import { render } from '@testing-library/react';
import TopComponent from './App';
import {initialStateTest} from './baselineState';
import ReactTestUtils from 'react-dom/test-utils'; // ES6


describe('<App />', () => {

  let App
  let mockStore
  let theStore

  beforeAll(() => {
    
  mockStore = configureMockStore()
  theStore=mockStore(initialStateTest)

  App = () => (
    <Provider store={theStore}>
        <TopComponent />
    </Provider>
)
  });
  

  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  /*
  the way children counting works is: you have your topmost render. 
  each <div> that has a subcompnoent called will be another child
  but also, if the topmost div *has its own content*, like a little text, it'll count
  so two subcomponent calls and hte text "this is the topmost div" = 3 children
  if you dont have any text in the top div, then it wont be a child, just teh two subcomponents

  thus, 2
  may haev to revise later if i put some things in the top component for some reason
  */
  
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});
it('renders correctly according to snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

});