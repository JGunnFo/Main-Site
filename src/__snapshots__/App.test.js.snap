// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="Overall-Grid-Parent"
>
  <div>
    <div>
      <div>
        <div>
          <div
            className="Main-Grid-Parent Main-Flex-Parent"
          >
            <div
              className="Main-Text-Emph Line1"
            >
              <img
                className="InlineL"
                src="transparent.png"
              />
              React Redux Front-End Developer.
            </div>
            <div
              className="Main-Text-Deemph Line2"
            >
              <img
                className="InlineL"
                src="transparent.png"
              />
              Jeffrey Gunn
            </div>
            <div
              className="Main-Text-Emph Line3"
            >
              <img
                className="InlineL"
                src="transparent.png"
              />
               Let's work together. [||]
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div
        className="Nav-Background"
      >
        <div
          className="Nav-Grid-Parent Nav-Flex-Parent"
        >
          <div>
            <div
              className="Nav-Unselected"
            >
              Résumé
            </div>
          </div>
          <div
            className="Nav-Unselected"
          >
            <div>
              <div>
                <button
                  onClick={[Function]}
                >
                  icon
                </button>
              </div>
              <div>
                Bio
              </div>
            </div>
          </div>
          <div
            className="Nav-Selected"
          >
            <div>
              <div>
                <button
                  onClick={[Function]}
                >
                  icon
                </button>
              </div>
              <div>
                Main
              </div>
            </div>
          </div>
          <div
            className="Nav-Unselected"
          >
            <div>
              <div>
                <button
                  onClick={[Function]}
                >
                  icon
                </button>
              </div>
              <div>
                Portfolio
              </div>
            </div>
          </div>
          <div
            className="Nav-Unselected"
          >
            <div>
              <div>
                <button
                  onClick={[Function]}
                >
                  icon
                </button>
              </div>
              <div>
                Github
              </div>
            </div>
          </div>
          <div
            className="Spacer"
          />
        </div>
         
      </div>
    </div>
  </div>
</div>
`;
